apps/costcontrol/test/unit/cost_control_test.js

```diff
@@ -498,4 +498,75 @@ suite('Cost Control Service Hub Suite >', function() {
       });
     }
   );
+
+  test(
+    'Global Data cache is refreshed when the resquest is not per App',
+    function(done) {
+      var expectedLastDataUsage = {
+        mobile: { total: 4800543137 },
+        wifi: { total: 112123944 }
+      };
+
+      var fakeSettings = { lastDataUsage: {}, lastDataUsagePerApp: {} };
+      this.sinon.stub(SimManager, 'requestDataSimIcc').yields({iccId: '12345'});
+      window.ConfigManager = new MockConfigManager({
+        fakeSettings: fakeSettings
+      });
+
+      CostControl.getInstance(function(service) {
+        service.request({type: 'datausage'}, function(result) {
+          done(function() {
+            assert.deepEqual(result.data.mobile.apps, {});
```

Comment:
Actually, it suffices with making 2 assertions, one to compare `result` with `CostControl.lastDataUsage`. This one would mean "the value is properly cached and can be retrieved without another call to `request()`". And the second one consist into see that `lastDataUsagePerApp` is not destroyed. For this, you can manually set `lastDataUsagePerApp` to a well known object and then see it it refers to the same, after the calling. What do you think?